app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomePresenter.kt

```diff
@@ -49,19 +55,43 @@ internal class HomePresenter(
 
         HomeEvent.MarkCardCrossSellsAsSeen -> {
           applicationScope.launch {
-            crossSellCardNotificationBadgeServiceProvider.provide().markAsSeen()
+            crossSellHomeNotificationServiceProvider.provide().markAsSeen()
           }
         }
+
+        is HomeEvent.CrossSellToolTipShown -> {
+          crossSellToolTipShownEpochDay = homeEvent.epochDay
+        }
+      }
+    }
+    LaunchedEffect(crossSellToolTipShownEpochDay) {
+      val epochDay = crossSellToolTipShownEpochDay
+      if (epochDay != null) {
+        crossSellHomeNotificationServiceProvider.provide().setLastEpochDayNewRecommendationNotificationWasShown(
+          epochDay,
+        )
       }
     }
-
     LaunchedEffect(loadIteration) {
       val forceNetworkFetch = loadIteration != 0
       Snapshot.withMutableSnapshot {
         isReloading = true
         hasError = false
       }
-      getHomeDataUseCaseProvider.provide().invoke(forceNetworkFetch).collectLatest { homeResult ->
+      combine(
+        getHomeDataUseCaseProvider.provide().invoke(forceNetworkFetch),
+        crossSellHomeNotificationServiceProvider.provide().showRedDotNotification(),
+        crossSellHomeNotificationServiceProvider.provide().getLastEpochDayNewRecommendationNotificationWasShown(),
+      ) { homeResult: Either<ApolloOperationError, HomeData>, showRedDot: Boolean, epochDay: Long? ->
+        homeResult to (showRedDot to epochDay)
+      }.collectLatest { result: Pair<Either<ApolloOperationError, HomeData>, Pair<Boolean, Long?>> ->
+        val homeResult = result.first
+        val showCrossSellRedDot = result.second.first
+        val lastDayCrossSellToolTipShown = result.second.second
```

Comment:
Creative 😅

```suggestion
        homeResult to (showRedDot to epochDay)
      }.collectLatest { (homeResult: Either<ApolloOperationError, HomeData>, (showCrossSellRedDot: Boolean, lastDayCrossSellToolTipShown: Long?)) ->
```

Could you check if this code also works?
Not sure if it's any better, but it's just a bit daunting reading the `result.second.first` and `result.second.second` down there :D